is_core/filters/default_filters.py

```diff
@@ -187,66 +221,174 @@ def get_widget(self, request):
+        return forms.Select(choices=(('', '-----'), (1, ugettext('Yes')), (0, ugettext('No'))))
 
 
-class NunberFieldFilter(DefaultFieldFilter):
+class NumberFieldFilter(DefaultFieldFilter):
 
     suffixes = ['gt', 'lt', 'gte', 'lte', 'in', 'isnull']
 
 
+class IntegerNumberFieldFilter(NumberFieldFilter):
+
+    def clean_value(self, value):
+        try:
+            return int(value)
+        except ValueError:
+            raise FilterValueException(ugettext('Value "{}" must be integer.'.format(value)))
+
+
+class FloatNumberFieldFilter(NumberFieldFilter):
+
+    def clean_value(self, value):
+        try:
+            return float(value)
+        except ValueError:
+            raise FilterValueException(ugettext('Value "{}" must be float.'.format(value)))
+
+
+class DecimalNumberFieldFilter(NumberFieldFilter):
+
+    def clean_value(self, value):
+        try:
+            return Decimal(value)
+        except InvalidOperation:
+            raise FilterValueException(ugettext('Value "{}" must be decimal.'.format(value)))
+
+
 class RelatedFieldFilter(DefaultFieldFilter):
-    pass
+
+    def get_last_rel_field(self, field):
+        if not field.is_relation:
+            return field
+        else:
+            next_field = field.rel.to._meta.get_field(field.rel.field_name)
+            return self.get_last_rel_field(next_field)
+
+
+class ForeignObjectRelFilter(RelatedFieldFilter):
+
+    def clean_value(self, value):
+        try:
+            return self.get_last_rel_field(self.field.field.model._meta.get_field(
+                self.field.field.model._meta.pk.name)).get_prep_value(value)
+        except ValueError:
+            raise FilterValueException(ugettext('Value "{}" is invalid.'.format(value)))
+
+
+class ForeignKeyFilter(RelatedFieldFilter):
+
+    def clean_value(self, value):
+        try:
+            return self.get_last_rel_field(self.field).get_prep_value(value)
+        except ValueError:
+            raise FilterValueException(ugettext('Value "{}" is invalid.'.format(value)))
+
+
+class ManyToManyFieldFilter(RelatedFieldFilter):
+
+    suffixes = ['all', 'in', 'isnull']
+
+    def clean_value(self, value):
+        try:
+            return get_last_rel_field(
+                self.field.rel.to._meta.get_field(self.field.m2m_target_field_name())
+            ).get_prep_value(value)
+        except ValueError:
+            raise FilterValueException(ugettext('Value "{}" is invalid.'.format(value)))
+
+    def clean_value_with_suffix(self, value, suffix):
+        if suffix == 'all':
+            return self._parse_list_values(value)
+        else:
+            return super(ManyToManyFieldFilter, self).clean_value_with_suffix(value, suffix)
+
+    def get_filter_term(self, value, suffix, request):
+        if suffix == 'all':
+            full_filter_key_without_suffix = self.full_filter_key.rsplit('__', 1)[0]
+            return Q(*(Q(**{full_filter_key_without_suffix: v}) for v in value))
```

Comment:
když už existuje in pro m2m který vrací objekt když obashuje alespon jeden z prvků uložených v poli musí existovat i all, který vrací objekt když obsahuje všechny prvky v poli